src/python/pants/core/goals/test.py

```diff
@@ -338,6 +357,11 @@ def open_coverage(self) -> bool:
         return cast(bool, self.options.open_coverage)
 
 
+@dataclass(frozen=True)
+class TestExtraEnv:
+    env: FrozenDict[str, Optional[str]]
```

Comment:
You can alternatively directly subclass `FrozenDict[str, Optional[str]]` and avoid the dataclass. I'm not sure what's better here - generally, I know composition is better than inheritance. But it's also often really nice to use inheritance, hence why we have `Collection` and `DeduplicatedCollection`.